opentelemetry-sdk/src/opentelemetry/sdk/propagator/b3_format.py

```diff
@@ -92,7 +93,11 @@ def extract(cls, get_from_carrier, carrier):
         if sampled in cls._SAMPLE_PROPAGATE_VALUES or flags == "1":
             options |= trace.TraceOptions.RECORDED
 
-        return trace.SpanContext(
+        # TODO: there is no standard way to retrieve
+        # the SpanContext or DistributedContext from a
+        # context object declared in opentelemetry-api.
+        # this should be more standard.
+        context[Tracer.CONTEXT_SLOT_NAME] = trace.SpanContext(
```

Comment:
I gave this PR a glance and I love where this is heading! Rather than passing the entire context to the propagator have we considered passing the slot to the constructor instead?

If the slot was passed, the slot `get`/`set` interface can be used to set the context, making the API more standard.

Reply:
the hope here was to give the the ability to the formatter to modify any context. I feel like if we pass explicit slots, then this would remove the flexibility.

But I think this discussion should probably continue in https://github.com/open-telemetry/opentelemetry-python/pull/89

Reply:
What are the use cases for a formatter modifying multiple context slots? And could those use cases be covered by creating multiple instances of the formatter with different slots in the constructor?

Reply:
Actually I think I understand the use-case, correct me if I'm wrong. Is this trying to cover the case where formats (to be defined) may transport span context and distributed context simultaneously?

While this doesn't cover the [named tracers](https://github.com/open-telemetry/opentelemetry-python/blob/ec0dfb42605ba34a0d2fd791378c7a93b6364398/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L287), I can see handling those cases as a separate PR. 👍

Reply:
Yep! Exactly. OTEP here: https://github.com/open-telemetry/oteps/pull/37

Yes, I'll pull up an issue on named tracers. I have a couple thoughts.

Reply:
here's my thoughts on the named tracer scenario: just creating multiple context objects instead. issue for discussion here: https://github.com/open-telemetry/opentelemetry-python/issues/126

Reply:
Named are AFAIK meant to solve a different set of use cases, e.g. disabling a troublesome part of instrumentation.

Reply:
> I feel like if we pass explicit slots, then this would remove the flexibility.

True, then we might as well go back to returning things (it feels like `sin(a: float, result_storage: SinusResultSlot) -> None`: just awkward).